fdm-calculator/src/norms/nl/2026/filling/input.test.ts

```diff
@@ -0,0 +1,167 @@
+    const mockFosfaatgebruiksnorm = 100
+
+    beforeEach(() => {
+        // Reset all mocks before each test
+        vi.clearAllMocks()
+
+        // Set up default mock implementations for a successful scenario
+        vi.mocked(getField).mockResolvedValue({
+            b_id: mockFieldId,
+            b_id_farm: "farm789",
+            b_centroid: [10, 20],
+            // Add other necessary Field properties
+        } as Field)
+        vi.mocked(getGrazingIntention).mockResolvedValue(true)
+        vi.mocked(isOrganicCertificationValid).mockResolvedValue(false)
+        vi.mocked(getCultivations).mockResolvedValue([
+            {
+                b_lu: "cult1",
+                b_lu_start: new Date(2026, 0, 1),
+                b_lu_end: new Date(2026, 5, 1),
+                b_lu_catalogue: "nl_2014",
+            },
+        ] as unknown as Cultivation[])
+        vi.mocked(getFertilizerApplications).mockResolvedValue([
+            { p_app_id: "app1", p_id_catalogue: "fert1", p_app_amount: 1000 },
+        ] as FertilizerApplication[])
+        vi.mocked(getFertilizers).mockResolvedValue([
+            { p_id_catalogue: "fert1", p_n_rt: 5, p_type_rvo: "115" },
+        ] as Fertilizer[])
+    })
+
+    it("should successfully collect all input data for a valid scenario", async () => {
+        const expectedB_centroid: [number, number] = [10, 20]
+        const expectedCultivations = [
+            {
+                b_lu: "cult1",
+                b_lu_start: new Date(2026, 0, 1),
+                b_lu_end: new Date(2026, 5, 1),
+                b_lu_catalogue: "nl_2014",
+            },
+        ]
+        const expectedApplications: FertilizerApplication[] = [
+            { p_app_id: "app1", p_id_catalogue: "fert1", p_app_amount: 1000 },
+        ]
+        const expectedFertilizers: Fertilizer[] = [
+            { p_id_catalogue: "fert1", p_n_rt: 5, p_type_rvo: "115" },
+        ]
+
+        const result = await collectNL2026InputForFertilizerApplicationFilling(
+            mockFdm,
+            mockPrincipalId,
+            mockFieldId,
+            mockFosfaatgebruiksnorm,
+        )
+
+        // Assert that all fdm-core functions were called with the correct arguments
+        expect(getField).toHaveBeenCalledWith(
+            mockFdm,
+            mockPrincipalId,
+            mockFieldId,
+        )
+        expect(getGrazingIntention).toHaveBeenCalledWith(
+            mockFdm,
+            mockPrincipalId,
+            "farm789",
+            2026,
+        )
+        expect(isOrganicCertificationValid).toHaveBeenCalledWith(
+            mockFdm,
+            mockPrincipalId,
+            "farm789",
+            new Date(2026, 4, 15),
+        )
+        expect(getCultivations).toHaveBeenCalledWith(
+            mockFdm,
+            mockPrincipalId,
+            mockFieldId,
+            {
+                start: new Date(2026, 0, 1),
+                end: new Date(2026, 11, 31, 23, 59, 59, 999),
+            },
```

Comment:
is this date by purpose set up to miliseconds?

Reply:
Yes to test the limit